src/React/DOM/Props.purs

```diff
@@ -638,6 +620,10 @@ onWheel :: forall eff props state result.
   (Event -> EventHandlerContext eff props state result) -> Props
 onWheel f = unsafeMkProps "onWheel" (handle f)
 
+ref :: forall eff props state result.
+  (Nullable Foreign -> EventHandlerContext eff props state result) -> Props
+ref f = unsafeMkProps "ref" (handle f)
```

Comment:
I've change `ref` to take a `Foreign`. `Ref` was a completely opaque data type, and the only way to validate it was by casting it to `Foreign` anyway.

Reply:
The reason is that I would like to avoid a dependency on `foreign` which pulls in a lot of other heavy dependencies. I think it's desirable to keep this package as light as we can, since it's a reasonable use case to use PureScript for one-off React components, in which case DCE can be important.

Reply:
If the user doesn't use ref functionality, won't the dependency be eliminated altogether in that case?

Reply:
Well _if_ they're using `purs bundle`, or if they're not using the React module as an entry point, then yes.

But I still don't see the reason to add a `foreign` dependency just for the `Foreign` type.

Reply:
It's not a hill I'm gonna die on, I just think it's weird to export an API with an opaque type and provide no way for working with it :P.

Reply:
Using `toForeign` isn't a big deal, but we should note it.